src/pages/task/taskTabs/FileTableTab/utils.ts

```diff
@@ -0,0 +1,23 @@
+import type { GroupedFiles } from "./types";
+
+/**
+ * `filterGroupedFiles` filters the groupedFiles array from the TaskFilesQuery
+ * @param groupedFiles - the groupedFiles array from the TaskFilesQuery
+ * @param search - the search string
+ * @returns - a new array of groupedFiles that contain the search string
+ */
+const filterGroupedFiles = (groupedFiles: GroupedFiles[], search: string) =>
+  groupedFiles.reduce((acc, groupedFile) => {
+    const filteredFiles = groupedFile?.files?.filter((file) =>
+      file?.name?.toLowerCase().includes(search.toLowerCase())
```

Comment:
(based on the schema, it looks like `name` will always be defined for a `file` if it exists)
```suggestion
      file.name.toLowerCase().includes(search.toLowerCase())
```